Issue with "Command line crash course": (Prettier tool install error)

MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Command_line

#### What information was incorrect, unhelpful, or incomplete?
I downloaded the latest 17.8.0 Current Node.js from the link they referenced, and from there they said to install Prettier by inputting this command on the terminal: npm install --global prettier

Instead of installing and running it gave this error:
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/prettier
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/prettier'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/prettier'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/lib/node_modules/prettier'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/victorrafael/.npm/_logs/2022-04-01T15_09_30_948Z-debug-0.log

I tried downloading the older 16.14.2 LTS Node.js and tried the same thing but it gave the same error. I'm a complete beginner with the command line and only know web development from the beginning of the "Learn web development" guide to up to this point. So I really don't know what to do or understand what people are saying when I copied and pasted the error on google.

For now, I guess I'll just skip the Prettier CLI tool introduction and I hope its not too important to the rest of learning web development. Thanks


#### Specific section or headline?


#### What did you expect to see?


#### Did you test this? If so, how?


<!-- Do not make changes below this line -->
<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/tools_and_testing/understanding_client-side_tools/command_line`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Command_line
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
* Last commit: https://github.com/mdn/content/commit/c4b6112d856dbb5fa7b929e5034499c605a51e08
* Document last modified: 2022-03-28T14:52:22.000Z

</details>

In this article ... , it says to install Prettier with: "npm install prettier -g" and that resulted in error messages but when I inputted: "npm install prettier" no errors showed up, but that also may be because I already installed it by inputting "npx prettier".

...
https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Introducing_complete_toolchain



1234567890-QWERTYUIOPasdfghjklZXCVBNMasdfghjkls-MacBook-Air:~ victorrafael$ npm install prettier -g
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/prettier
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/prettier'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/prettier'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/lib/node_modules/prettier'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/victorrafael/.npm/_logs/2022-04-07T16_49_50_419Z-debug-0.log
1234567890-QWERTYUIOPasdfghjklZXCVBNMasdfghjkls-MacBook-Air:~ victorrafael$ npm install prettier

up to date, audited 2 packages in 581ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
1234567890-QWERTYUIOPasdfghjklZXCVBNMasdfghjkls-MacBook-Air:~ victorrafael$

--- a/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
+++ b/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
@@ -396,21 +396,7 @@ For this article we will install Prettier locally, as suggested in the [Prettier
 Once you've installed node, open up the terminal and run the following command to install Prettier:
 
 ```bash
-npm install --save-dev --save-exact prettier
-```
-
-Then create an empty `.prettierrc.json` config file to let editors and other tools know you are using Prettier:
-
-```bash
-echo {}> .prettierrc.json
-```
-
-You can optionally create a file named `.prettierignore` to specify files and directories that you definitely don't want to format.
-This should be created in the same folder and has exactly the same format as `.gitignore`.
-For example, if you don't want to format HTML files, the file might contain this line:
-
-```
-*.html
+npm install prettier
 ```
 
 You can now run the file locally using the [npx](https://nodejs.dev/learn/the-npx-nodejs-package-runner) tool.

--- a/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
+++ b/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
@@ -327,7 +327,7 @@ Install npm on your system now, by going to the URL above and downloading and ru
 ![the node.js installer on windows, showing the option to include npm](npm-install-option.png)
 
 Although we'll look at a number of different tools in the next article onwards, we'll cut our teeth on [Prettier](https://prettier.io/).
-Prettier is an opinionated code formatter that has "few options".
+Prettier is an opinionated code formatter that only has a "few options".
 Fewer options tends to mean simpler.
 Given how tooling can sometimes get out of hand in terms of complexity, "few options" can be very appealing.
 
@@ -376,13 +376,14 @@ There's pros and cons each way — and this list of pros and cons for globally i
   </tbody>
 </table>
 
-Although the _cons_ list is shorter, the negative impact of global installing is potentially much larger than the benefits. However, for now we'll err on the side of simplicity and install globally to keep things simple. We'll look more at local installs and why they're good in the next article.
+Although the _cons_ list is shorter, the negative impact of global installing is potentially much larger than the benefits.
+Here we'll install locally, but feel free to install globally once you understand the relative risks.
+
 
 ### Installing Prettier
 
-For this article we will install Prettier as a global command line utility.
-
 Prettier is an opinionated code formatting tool for front end developers, focusing around JavaScript-based languages and adding support for HTML, CSS, SCSS, JSON and more.
+
 Prettier can:
 
 - Save the cognitive overhead of getting the style consistent manually across all your code files; Prettier can do this for you automatically.
@@ -390,24 +391,34 @@ Prettier can:
 - Be installed on any operating system and even as a direct part of project tooling, ensuring that colleagues and friends who work on your code use the code style you're using.
 - Be configured to run upon save, as you type, or even before publishing your code (with additional tooling that we'll see later on in the module).
 
+For this article we will install Prettier locally, as suggested in the [Prettier installation guide](https://prettier.io/docs/en/install.html)
+
 Once you've installed node, open up the terminal and run the following command to install Prettier:
 
 ```bash
-npm install --global prettier
+npm install --save-dev --save-exact prettier
 ```
 
-> **Note:** If you get "permission denied" errors on Linux when running the above command, you can try installing with `sudo`:
-> ```bash
-> sudo -H npm install --global prettier
-> ```
-
-
-Once the command has finished running, the Prettier tool is now available in your terminal, at any location in your file system.
-
-Running the command without any arguments, as with many other commands, will offer up usage and help information. Try this now:
+Then create an empty `.prettierrc.json` config file to let editors and other tools know you are using Prettier:
 
 ```bash
-prettier
+echo {}> .prettierrc.json
+```
+
+You can optionally create a file named `.prettierignore` to specify files and directories that you definitely don't want to format.
+This should be created in the same folder and has exactly the same format as `.gitignore`.
+For example, if you don't want to format HTML files, the file might contain this line:
+
+```
+*.html
+```
+
+You can now run the file locally using the [npx](https://nodejs.dev/learn/the-npx-nodejs-package-runner) tool.
+Running the command without any arguments, as with many other commands, will offer up usage and help information.
+Try this now:
+
+```bash
+npx prettier
 ```
 
 Your output should look something like this:
@@ -421,7 +432,13 @@ Stdin is read if it is piped to Prettier and no files are given.
 …
 ```
 
-It's always worth at the very least skimming over the usage information, even if it is long. It'll help you to understand better how the tool is intended to be used.
+It's always worth at the very least skimming over the usage information, even if it is long.
+It'll help you to understand better how the tool is intended to be used.
+
+> **Note:** If you have not first installed prettier locally, then running `npx prettier` will download and run the latest version of prettier all in one go _just for that command_.
+> While that might sound great, new versions of prettier may slightly modify the output.
+> You want to install locally so that you are fixing the version of prettier that you are using for formatting until you are ready to change it.
+
 
 ### Playing with Prettier
 
@@ -441,10 +458,10 @@ printMe(myObj)
 We can run prettier against a codebase to just check if our code wants adjusting. `cd` into your directory, and try running this command:
 
 ```bash
-prettier --check index.js
+npx prettier --check index.js
 ```
 
-You should get on output along the lines of
+You should get on output along the lines of:
 
 ```bash
 Checking formatting...
@@ -457,7 +474,7 @@ So there's some code styles that can be fixed. No problem. Adding the `--write` 
 Now try running this version of the command:
 
 ```bash
-prettier --write index.js
+npx prettier --write index.js
 ```
 
 You'll get an output like this

--- a/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
+++ b/files/en-us/learn/tools_and_testing/understanding_client-side_tools/command_line/index.md
@@ -21,15 +21,13 @@ In your development process you'll undoubtedly be required to run some command i
       <th scope="row">Prerequisites:</th>
       <td>
         Familiarity with the core <a href="/en-US/docs/Learn/HTML">HTML</a>,
-        <a href="/en-US/docs/Learn/CSS">CSS</a>, and
-        <a href="/en-US/docs/Learn/JavaScript">JavaScript</a> languages.
+        <a href="/en-US/docs/Learn/CSS">CSS</a>, and <a href="/en-US/docs/Learn/JavaScript">JavaScript</a> languages.
       </td>
     </tr>
     <tr>
       <th scope="row">Objective:</th>
       <td>
-        To understand what the terminal/command line is, what basic commands you
-        should learn, and how to install new command line tools.
+        To understand what the terminal/command line is, what basic commands you should learn, and how to install new command line tools.
       </td>
     </tr>
   </tbody>
@@ -356,21 +354,20 @@ There's pros and cons each way — and this list of pros and cons for globally i
     <tr>
       <td>Only install once</td>
       <td>
-        Other developers in your team won't have access to these tools, for
-        example if you are sharing the codebase over a tool like git.
+        Other developers in your team won't have access to these tools, for example if you are sharing the codebase over a tool like git.
       </td>
     </tr>
     <tr>
       <td>Uses less disk space</td>
       <td>
-        Related to the previous point, it makes project code harder to replicate
-        (if you install your tools locally, they can be set up as dependencies
-        and installed with <code>npm install</code>).
+        Related to the previous point, it makes project code harder to replicate (if you install your tools locally, they can be set up as dependencies and installed with <code>npm install</code>).
       </td>
     </tr>
     <tr>
       <td>Always the same version</td>
-      <td></td>
+      <td>
+        Global installation can fail with "permission denied" errors, particularly on Linux which is protective about process and user access to the file system.
+        Resolving these errors usually requires Linux expertise, or granting more permission than you might want to the global process (for example, using <code>sudo</code>).</td>
     </tr>
     <tr>
       <td>Feels like any other unix command</td>
@@ -398,6 +395,12 @@ Once you've installed node, open up the terminal and run the following command t
 ```bash
 npm install --global prettier
 ```
+
+> **Note:** If you get "permission denied" errors on Linux when running the above command, you can try installing with `sudo`:
+> ```bash
+> sudo -H npm install --global prettier
+> ```
+
 
 Once the command has finished running, the Prettier tool is now available in your terminal, at any location in your file system.
 

